tests/unit/s2n_client_empty_cert_test.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <stdint.h>
+#include <s2n.h>
+
+#include "s2n_test.h"
+
+#include "tls/s2n_tls.h"
+#include "tls/s2n_connection.h"
+#include "utils/s2n_safety.h"
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* Test s2n_send_empty_cert_chain */
+    {
+        struct s2n_stuffer out;
+        EXPECT_SUCCESS(s2n_stuffer_alloc(&out, 3));
+
+        EXPECT_SUCCESS(s2n_send_empty_cert_chain(&out));
+        EXPECT_EQUAL(s2n_stuffer_data_available(&out), 3);
+        uint32_t cert_len;
+        EXPECT_SUCCESS(s2n_stuffer_read_uint24(&out, &cert_len));
+        EXPECT_EQUAL(cert_len, 0);
+
+        EXPECT_SUCCESS(s2n_stuffer_free(&out));
+    }
+
+    /* Client sends the empty cert when no client default chain and key */
+    {
+        struct s2n_config *client_config;
+        EXPECT_NOT_NULL(client_config = s2n_config_new());
+
+        struct s2n_connection *client_conn;
+        EXPECT_NOT_NULL(client_conn = s2n_connection_new(S2N_CLIENT));
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, client_config));
+        EXPECT_SUCCESS(s2n_connection_set_client_auth_type(client_conn, S2N_CERT_AUTH_OPTIONAL));
+
+        /* client send empty cert */
+        EXPECT_SUCCESS(s2n_client_cert_send(client_conn));
```

Comment:
Do we need to add a test that expects failure and inability to send empty Certificate message when this is set to S2N_CERT_AUTH_REQUIRED or is that covered in another set of unit tests?

Reply:
i dont know if it is added anywhere but it's easy enough. i'll just add it here